### PR TITLE
Updated flapdoodle embedded to 2.0

### PR DESCRIPTION
**Pull request by kblief**
Updated the flapdoodle library to 2.0.0
Added the flapdoodle embedded process dependency
Upped the build number
Updated the tests

### CODE REVIEW

File: src/test/groovy/org/grails/plugin/embedded/mongodb/EmbeddedMongoDBGrailsPluginSpec.groovy [hunk 1]
```diff
@@ -27,24 +27,24 @@ class EmbeddedMongoDBGrailsPluginSpec extends Specification {
...
         expect:
-        new EmbeddedMongoDBGrailsPlugin(grailsApplication: grailsApplication).getVersion() == Version.V3_2_1
+        new EmbeddedMongoDBGrailsPlugin(grailsApplication: grailsApplication).getVersion() == Version.V3_4_1
     }
 
     void "test version release candidate"() {
         given:
         GrailsApplication grailsApplication = Mock(GrailsApplication) {
             1 * getConfig() >> {
-                new PropertySourcesConfig([grails: [mongodb: [version: "3.2.1-rc3"]]])
```
Comment (jameskleeh) [inline src/test/groovy/org/grails/plugin/embedded/mongodb/EmbeddedMongoDBGrailsPluginSpec.groovy]: The purpose of this test is to verify it parses a RC release. Your change is not meeting that goal

File: build.gradle [hunk 2]
```diff
@@ -50,14 +50,15 @@ dependencies {
...
+    compile("de.flapdoodle.embed:de.flapdoodle.embed.mongo:2.0.0") {
         exclude group: "commons-io"
         exclude group: "org.apache.commons"
         exclude group: "net.java.dev.jna"
         exclude group: "org.slf4j"
     }
-    provided "org.grails.plugins:mongodb:5.0.8"
-    documentation "org.codehaus.groovy:groovy-all:2.4.6"
+    compile "de.flapdoodle.embed:de.flapdoodle.embed.process:2.0.1"
+    provided "org.grails.plugins:mongodb:6.0.11"
```
Comment (jameskleeh) [inline build.gradle]: Why update the mongodb plugin?

File: src/test/groovy/org/grails/plugin/embedded/mongodb/EmbeddedMongoDBGrailsPluginSpec.groovy [hunk 3]
```diff
@@ -27,24 +27,24 @@ class EmbeddedMongoDBGrailsPluginSpec extends Specification {
...
+                new PropertySourcesConfig([grails: [mongodb: [version: "3.5.1"]]])
```
Comment (jameskleeh) [inline src/test/groovy/org/grails/plugin/embedded/mongodb/EmbeddedMongoDBGrailsPluginSpec.groovy]: The purpose of this test is to verify it parses a RC release. Your change is not meeting that goal

Comment (kblief) [inline src/test/groovy/org/grails/plugin/embedded/mongodb/EmbeddedMongoDBGrailsPluginSpec.groovy]: Agreed, re-added test.

File: build.gradle (hunk 2 above)
Comment (kblief) [inline build.gradle]: Trying to stay current, changed back to previous version.

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 language: groovy
 sudo: false
 jdk:
-- openjdk7
+- oraclejdk8
```
Comment (jameskleeh) [inline .travis.yml]: Why did you change this? The plugin needs to work with Grails which is built with Java 7

**Comment by jameskleeh** [comment]
"Starting with version 2 we are going to support only java 8 or higher". Requiring Java 8 is a breaking change and would require a major version change in this library. Probably would make sense to update this to 2.0.0, which would require creating a 1.x branch.

File: .travis.yml (hunk 4 above)
Comment (kblief) [inline .travis.yml]: the flapdoodle 2.0 is built with jdk8.
